src/packet_analysis/protocol/tcp/TCP.cc

```diff
@@ -125,7 +125,7 @@ void TCPAnalyzer::DeliverPacket(Connection* c, double t, bool is_orig, int remai
 	pkt->session = c;
 
 	// Send the packet back into the packet analysis framework.
-	ForwardPacket(std::min(len, remaining), data, pkt);
+	ForwardPacket(len, data, pkt);
```

Comment:
The reason this breaks is because we take the `len` value passed into `GeneveAnalyzer::AnalyzePacket` and use it as the length value in the inner packet object. When it gets down in `IPAnalyzer::AnalyzePacket` it tries to compare that length value to the length in the actual IP header, sees that the one in the packet is too low, generates a weird, and bails out of processing any of the rest of the inner packet.

After reverting the above change, for example, in the first packet in geneve-vxlan-dns-truncated.pcap, `packet->len` ends up as 126 where the IP header says it should be 132 (which matches the len vs remaining difference).

We could fix `IPTunnel::build_inner_packet` to do some detection on whether it's IP and set the length based on that etc etc, but I think we'd be right back where we were with lying about how much data is actually remaining again.

Reply:
Yeah, that looks good to me. Should I close this PR and open one against that? @ckreibich will have to approve it since we've both touched this one now.

Does this fix the asan crash with the other pcap?

Reply:
> Does this fix the asan crash with the other pcap?

You mean the one from the following issue?

https://github.com/zeek/zeek/issues/2683 

Yep, seems fine. I'm adding it as a btest. I'll rebase and create a new PR. I'm going to squash the fixup commit.

Reply:
Nah I meant the geneve-truncated pcap that Cirrus was having problems with on this branch. Looks like it's fine with the other branch.